juju/juju
# Simplify Dqlite App Construction

- Moves REPL start-up from the Dqlite app wrapper to the worker.
- Removes the CGo/not-CGo hedging - we can assume we will always build with CGo.
- Removes our superfluous app wrapper, opting for direct instantiation of the upstream reference.
- Fixes spelling of "retryable".
- Removes ACL option from s3 statements for pushing built Dqlite dependencies - the bucket now uses policies instead.

## QA steps

- `make go-install`.
- Bootstrap.
- `juju ssh -m controller 0`.
- `juju_engine_report | grep -A 6 db-accessor`.
- We should see that the worker is running:
```
  db-accessor:
    inputs:
    - agent
    - is-controller-flag
    start-count: 1
    started: "2022-09-06 11:22:29"
    state: started
```

## Documentation changes

None.

## Bug reference

N/A
